File: examples/0_Empty/main.cpp
#include "../../TinyEngine.h"

int main( int argc, char* args[] ) {

	//Initialize a Window
	Tiny::window("Example Window", 600, 400);

	//Add the Event Handler
	Tiny::event.handler = [&](){
	};

	//Set up an ImGUI Interface here
	Tiny::view.interface = [&](){
	};

	//Define the rendering pipeline
	Tiny::view.pipeline = [&](){
	};

	//Execute the render loop
	Tiny::loop([&](){
		//Do any additional stuff in here
	});

	Tiny::quit();

	return 0;
}

File: examples/0_Empty/main.py
from tinyengine import tiny

# Main Functions

def loop():
    return;

def pipeline():
    tiny.view.target([1.0, 1.0, 1.0], True)

def interface():
    return;

def handler():
    return;

# Main Loop

tiny.window("Example Window", 600, 400)

tiny.view.pipeline(pipeline);
tiny.view.interface(interface);
tiny.event.handler(handler);

tiny.loop(loop);
tiny.quit()
